Automattic/wp-calypso
# Store: Add refunds to the orders activity log

Fixes #21975 (kind of) – This PR adds all refunds to the activity log, so now this list includes manual refunds. It _also_ includes duplicates of "API refunds" (refunds done via Stripe/PayPal/other API payment system), this is because there isn't a way to differentiate these from "manual refunds" in the API response.

So now the Order's Activity Log looks like this:

<img width="865" alt="screen shot 2018-04-24 at 3 46 30 pm" src="https://user-images.githubusercontent.com/541093/39210505-ee0ac962-47d6-11e8-996d-11d73258a0fc.png">

That `0.00` refund was a mistake but serves as an example of a manual refund without "reason" text; the `25.00` refund is an API refund with a "reason", and you can see the stripe transaction note above it. I used the credit card icon for a "money" feeling 🙂

We could probably update the API response in wc-api-dev to return some distinction between manual & api refunds, if we want to.

On the technical side, this required adding a new reducer & fetching data from a new endpoint – the refunds included in the order object aren't detailed enough, so we need to request [the full refund objects](https://woocommerce.github.io/woocommerce-rest-api-docs/#order-refund-properties) from the `/orders/:order_id/refund` endpoint.

This means I've updated the refunds state & data layers, and added tests for all of the changes.

**To test**

- View an order with refunds, or add refunds to an order: `/store/order/:site/:order_id`
- You should see your refunds in the Activity Log section of the order, on the date of the refund.

You can also run the tests to make sure my new tests pass 🙂 `npm run test-client client/extensions/woocommerce/state`
